### ISSUE TITLE
Need a database upgrade

### ISSUE THREAD

**Issue by scottkoland** [roadkillwiki/roadkill#65]
Hello,

I have a roadkill wiki sqlite database that needs to be upgraded to 2.0. The url in the upgrade steps says to post a request on the Roadkill Wiki Google Groups (which is now closed) or send a message to bitbucket user Google of the bitbucket user mrshrinkray (which is now a dead link).

How can I upgrade my sqlite database?

Thanks,
Scott

**Comment by yetanotherchris** [comment]
Unfortunately 1.7 isn't supported anymore, but it is still possible if you're willing to manually change the schema.

The prefered option would be to migreate to mySQL, Postgres or SQL Server and then upgrade. Would you be able to do this?

**Comment by scottkoland** [comment]
Hi Chris,
Yes, we could switch to SQL Server. What is the best way to migrate from our sqlite database to SQL Server or another of these formats?

**Comment by yetanotherchris** [comment]
http://www.sqlite.org/cvstrac/wiki?p=ConverterTools has a number of tools, maybe SQLite Data Wizard?

**Comment by scottkoland** [comment]
Thanks. I'm taking a look at SQLite Data Wizard. Is there a good process to follow to migrate the Roadkill wiki database schema from sqlite to Sql Server?

**Comment by yetanotherchris** [comment]
The column names are all the same so it should be easy to do, and there's only four tables. Hopefully the tool will create the keys for you in SQL Server, but they're just simple UniqueIdentifier Primary keys.

**Comment by scottkoland** [comment]
Thanks Chris. I was able to import the data from roadkill_pages and roadkill_pagecontent, using DB Browser to export and MySQL Workbench to import to the new MySQL. I wasn't able to get the users and siteconfiguration tables to import properly, but the siteconfiguration is a different format now and recreating the users for the wiki weren't a big issue for me.

I installed the new 2.0 site then let the app create the db following the wiki setup, then I inserted the pages and pagecontent via MySQL Workbench. Worked pretty well, but on MySQL on Azure, the page Id and pagecontent index numbers got a bit out of whack, so I had to modify some Ids in pagecontext. All in all, not too bad. And now it's up and running with 2.0.
